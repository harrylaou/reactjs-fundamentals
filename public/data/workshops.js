[
  {
    "title": "React Router and fetch data",
    "url": "http://www.meetup.com/JS-Lab-London/events/231907561/",
    "students" : "Unlimited",
    "price": "Free",
    "instructors": [
        {
            "name":"Alex Lobera",
            "url": "http://alexlobera.com/",
            "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses.jpg"
        }
    ]
  },
  {
    "title": "React Master class",
    "url": "http://techdency.com/#react-training",
    "price": "Â£ 595",
    "students" : "10 per instructor",
    "instructors": [
        {
            "name":"Alex Lobera",
            "url": "http://alexlobera.com/",
            "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses.jpg"
        },
        {
            "name":"Davy Engone",
            "url": "https://github.com/davyengone",
            "avatar": "https://pbs.twimg.com/profile_images/589582545022545921/qQRHggUr.jpg"
        }
    ]
  }
]