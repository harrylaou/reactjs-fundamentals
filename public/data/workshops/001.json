{
  "id": "001",
  "title": "React Router and fetch data",
  "url": "http://www.meetup.com/JS-Lab-London/events/231907561/",
  "students" : "Unlimited",
  "price": "Free",
  "instructors": [
    {
      "name":"Alex Lobera",
      "url": "http://alexlobera.com/",
      "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses_400x400.jpg"
    }
  ]
}