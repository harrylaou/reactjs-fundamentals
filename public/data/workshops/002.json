{
  "id": "002",
  "title": "React Master class",
  "url": "http://techdency.com/#react-training",
  "price": "Â£ 595",
  "students" : "10 per instructor",
  "instructors": [
    {
      "name":"Alex Lobera",
      "url": "http://alexlobera.com/",
      "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses_400x400.jpg"
    },
    {
      "name":"Mark Rivelles",
      "url": "https://github.com/fsdevlondon",
      "avatar": "https://pbs.twimg.com/profile_images/784142672370003969/Hj51EXIt.jpg"
    }
  ]
}