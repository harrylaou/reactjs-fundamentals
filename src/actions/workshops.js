export const RECEIVE_WORKSHOPS = 'RECEIVE_WORKSHOPS'

export const receiveWorkshops = workshops => ({
  type: RECEIVE_WORKSHOPS,
  workshops
})
