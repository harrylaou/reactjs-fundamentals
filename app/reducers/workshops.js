import {RECEIVE_WORKSHOPS} from '../actions/workshops'

const workshops = (state = [], action) => {
  switch (action.type) {
    case RECEIVE_WORKSHOPS: {
      return action.workshops
    }
    default:
      return state
  }
}

export default workshops
