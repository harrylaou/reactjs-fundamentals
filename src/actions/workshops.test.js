import {receiveWorkshops, RECEIVE_WORKSHOPS} from './workshops'

test(`Return an Action tyoe ${RECEIVE_WORKSHOPS}`, () => {
  const workshops = ["Reactjs Fundamentals"]
  expect(receiveWorkshops(workshops)).toEqual({
    type: RECEIVE_WORKSHOPS,
    workshops
  })

  expect(receiveWorkshops(workshops)).toMatchSnapshot()
})