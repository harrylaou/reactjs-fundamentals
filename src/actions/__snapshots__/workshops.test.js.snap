// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Return an Action tyoe RECEIVE_WORKSHOPS 1`] = `
Object {
  "type": "RECEIVE_WORKSHOPS",
  "workshops": Array [
    "Reactjs Fundamentals",
  ],
}
`;
